Backport of:

From 1d59a686ff8c429a3347d9d30c49627d47169fdc Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Mon, 15 May 2017 21:10:19 +0200
Subject: [PATCH] Added check to prevent image being 0x0 (reported in #489).

A crafted file revealed an assertion failure in profile.c.

magick: MagickCore/profile.c:1303: ResetImageProfileIterator: Assertion `image != (Image *) ((void *)0)' failed.

origin: https://github.com/ImageMagick/ImageMagick/commit/f5910e91b0778e03ded45b9022be8eb8f77942cd
bug: https://github.com/ImageMagick/ImageMagick/issues/489
---
 coders/dds.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

Index: imagemagick-6.7.7.10/coders/dds.c
===================================================================
--- imagemagick-6.7.7.10.orig/coders/dds.c	2017-05-26 09:06:07.656686135 -0400
+++ imagemagick-6.7.7.10/coders/dds.c	2017-05-26 09:07:01.357370122 -0400
@@ -359,7 +359,10 @@ static Image *ReadDDSImage(const ImageIn
   
   if (volume)
     num_images = dds_info.depth;
-  
+
+  if (num_images < 1)
+    ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+
   for (n = 0; n < num_images; n++)
   {
     if (n != 0)
